clover.rb

```diff
@@ -342,19 +348,41 @@ def assets
     password_confirm_label "Password Confirmation"
     before_create_account do
       Validation.validate_cloudflare_turnstile(param("cf-turnstile-response"))
-      account[:id] = Account.generate_uuid
-      account[:name] = param("name")
-      Validation.validate_account_name(account[:name])
+      scope.before_rodauth_create_account(account, param("name"))
     end
     after_create_account do
-      account = Account[account_id]
-      account.create_project_with_default_policy("Default")
-      ProjectInvitation.where(email: account.email).each do |inv|
-        account.associate_with_project(inv.project)
-        if (managed_policy = Authorization::ManagedPolicy.from_name(inv.policy))
-          managed_policy.apply(inv.project, [account], append: true)
+      scope.after_rodauth_create_account(account_id)
+    end
+
+    # :nocov:
+    if Config.omniauth_github_id
+      require "omniauth-github"
+      omniauth_provider :github, Config.omniauth_github_id, Config.omniauth_github_secret
+    end
+    if Config.omniauth_google_id
+      require "omniauth-google-oauth2"
+      omniauth_provider :google_oauth2, Config.omniauth_google_id, Config.omniauth_google_secret, name: :google
+    end
+    # :nocov:
+
+    before_omniauth_create_account do
+      scope.before_rodauth_create_account(account, omniauth_name)
+    end
+
+    after_omniauth_create_account do
+      scope.after_rodauth_create_account(account_id)
+    end
+
+    before_omniauth_callback_route do
+      account = Account[account_from_omniauth&.dig(:id)]
```

Comment:
```suggestion
      account = Account[account_from_omniauth&.[](:id)]
```

It's not necessary to use `dig` for a single attribute.

Reply:
I remembered you used `&` and `[]` together in one of your PRs, but I couldn't find the example and couldn't use `[]` with `&`. Thanks for that; I applied it